packages/web3/src/entitlement.ts

```diff
@@ -18,6 +10,12 @@ import { MOCK_ADDRESS } from './Utils'
 
 const zeroAddress = ethers.constants.AddressZero
 
+// This public client is used purely to infer the typescript type of RuleData.
```

Comment:
@texuf if you know a better way here? I took a quick look at ethers docs but nothing jumped out.

Reply:
Can you try the ExtractAbi function that @shuhuiluo used here: https://github.com/river-build/river/pull/675/files

Reply:
We actually don't need the function, just the type here. I found another approach, PTAL?